src/generators/clientContextFileGenerator.ts

```diff
@@ -87,31 +90,34 @@ function writeClassProperties(
   clientParams: ParameterDetails[]
 ) {
   contextClass.addProperties(
-    clientParams.map(param => {
-      return {
-        name: param.name,
-        type: param.typeDetails.typeName,
-        hasQuestionToken: !param.required
-      } as PropertyDeclarationStructure;
-    })
+    clientParams
+      .filter(p => !p.isSynthetic)
```

Comment:
What is a synthetic parameter?

Reply:
I've been using Synthetic parameters when they are not defined in the Swagger file, for example, I'm injecting the Credentials parameter as a Synthetic one.

The reason for doing this is to just add the parameter and let it flow through the transform /generation process so we don't need to it handle specially in the layers above. 

Before having Credentials as synthetic, we used to have a specific check on the Generator layer to write the code for the credentials parameter.

Reply:
Thanks for that explanation!